dotnet-authserver/src/TeacherIdentity.AuthServer/Pages/Admin/Users.cshtml.cs

```diff
@@ -117,4 +122,57 @@ public record UserInfo
         public required string? Trn { get; init; }
         public required TrnLookupStatus? TrnLookupStatus { get; init; }
     }
+
+    private async Task<Expression<Func<User, bool>>> GetFilterPredicate(TrnLookupStatus?[] lookupStatus, string? userSearch)
+    {
+        var filterPredicate = PredicateBuilder.New<User>(user => user.UserType == UserType.Default);
+
+        if (lookupStatus.Length > 0)
+        {
+            filterPredicate.And(user => lookupStatus.Contains(user.TrnLookupStatus));
+        }
+
+        if (!string.IsNullOrEmpty(userSearch))
+        {
+            switch (userSearch)
+            {
+                case var searchString when new Regex(@"^\d{7}$").IsMatch(searchString):
+                    filterPredicate.And(user => user.Trn == searchString);
```

Comment:
Andrew mentioned these could also be searched for via UserSearchAttributes. Might be better performance wise, but not sure if it would read better. How concerned are we about performance?

Reply:
There's a unique index on `trn` so I suspect this is easily fast enough.